src/App.tsx

```diff
@@ -14,37 +15,110 @@
   'Jam',
   'Garlic',
 ];
+enum SortType {
+  BY_ALPH = 'alph',
+  BY_LENGTH = 'length',
+  DEFAULT = '',
+}
+
+type FilterProps = {
+  sortField: SortType;
+  reverse: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, reverse }: FilterProps,
+) {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods = preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.BY_ALPH:
+          return good1.localeCompare(good2);
+
+        case SortType.BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.DEFAULT);
+  const [reverse, setReverse] = useState(false);
+
+  const goods = getPreparedGoods(goodsFromServer, { sortField, reverse });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button is-info', {
+            'is-light': sortField !== SortType.BY_ALPH,
+          })}
+          onClick={() => {
```

Comment:
[UX] - Don't forget to mark the active button to explain to the user what action is active. It seems like you are already doing this by toggling the 'is-light' class based on the sortField state, which is good. Ensure that this UX pattern is consistently applied across similar elements.